### DISCUSSION TITLE
Deploy via ImageName

### DISCUSSION THREAD

**Discussion by pcace** [caprover/caprover#1602] [Q&A]
Hi, do i understand correctly that i can use Caprover to automatically look for new images i have pushed to ghcr.io? 
If so, what is the way to get rid of this error Message: 

```
------------------------- Fri Dec 16 2022 23:09:34 GMT+0000 (Coordinated Universal Time)
Build started for staging
An explicit image name was provided (ghcr.io/user/redwood-docker-compose-web:latest). Therefore, no build process is needed.
Pulling this image: ghcr.io/user/redwood-docker-compose-web:latest This process might take a few minutes.
Build has failed!
----------------------
Deploy failed!
Error: (HTTP code 500) server error - Head "https://ghcr.io/v2/user/redwood-docker-compose-web/manifests/latest": unauthorized

```

thanks a lot!

**Comment by githubsaturn** [comment]
As you can see, the error is an authorization issue. You'll have to add your registry to your instance. 

Go to cluster and add your registry to the Docker registries. After adding the registry, make sure to mark Default Push Registry as "DISABLE".

![image](https://user-images.githubusercontent.com/10582082/208227806-4e08480b-2a99-4861-abae-8d688cbcf8b8.png)


Make sure to use a [Personal Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic) as password.

**Reply by pcace** [reply]
hi, thanks a lot! that seems to work. 
but does this always trigger a rebuild once there is a new image or how would caprover know it needs to rebuild?

Cheers

**Reply by githubsaturn** [reply]
No, this is a one time deploy. You can use github actions if you want to deploy on build:
See https://github.com/caprover/deploy-from-github

**Reply by pcace** [reply]
hi, thank you so much for your help! slowly i am getting there ;) 
i just ran into an other problem, so i opened up another issue, hope thats ok (https://github.com/caprover/caprover/issues/1605)
